js-core/homeworks/homework-5/src/main.js

```diff
@@ -0,0 +1,201 @@
+/*
+ 1. Переместите 0 в конец массива, остальные числа должны остаться
+ неизменными
+ // concat
+ example:
+ [1,false,2,0,3,null,0,4,0,25] => [1, false, 2, 3, null, 4, 25, 0, 0, 0]
+ [ 'a', 0, 0, 'b', null, 'c', 'd', 0, 1, false, 0, 1, 0, 3, [], 0, 1, 9, 0, 0, {}, 0, 0, 9 ] => ["a","b",null,"c","d",1,false,1,3,[],1,9,{},9,0,0,0,0,0,0,0,0,0,0]
+ [ 0, 1, null, 2, false, 1, 0 ] => [1,null,2,false,1,0,0]
+ */
+
+let arr1 = [1, false, 2, 0, 3, null, 0, 4, 0, 25];
+let arr2 = [
+  'a',
+  0,
+  0,
+  'b',
+  null,
+  'c',
+  'd',
+  0,
+  1,
+  false,
+  0,
+  1,
+  0,
+  3,
+  [],
+  0,
+  1,
+  9,
+  0,
+  0,
+  {},
+  0,
+  0,
+  9
+];
+
+function moveZeroToEnd(arr) {
+	let zeroArray = [];
+	
+	while(arr.indexOf(0)!=-1){	
```

Comment:
it's not really easy to understand what's going on here and if in some moment we will need to extend such function it will be hard to do.

How would you solve that problem if you would think about readability of code ?